src/main.rs

```diff
@@ -77,6 +77,28 @@ fn main() -> glib::ExitCode {
                 application_opts.test_dc = true;
             }
 
+            application_opts.client_id = config::TG_API_ID;
+            application_opts.client_secret = config::TG_API_HASH.to_string();
+
+            match dict.lookup::<String>("telegram-client-id") {
+                Ok(value) => match value {
+                    Some(client_id) => match client_id.parse::<i32>() {
+                        Ok(client_id_parsed) => application_opts.client_id = client_id_parsed,
+                        Err(err) => log::error!("Invalid value for client id: {err}"),
+                    },
+                    None => log::warn!("Could not extract value!"),
+                },
+                Err(err) => log::warn!("Could not extract value: {err}"),
+            };
+
+            match dict.lookup::<String>("telegram-client-secret") {
+                Ok(value) => match value {
+                    Some(client_secret) => application_opts.client_secret = client_secret,
+                    None => log::warn!("Could not extract secret!"),
+                },
+                Err(err) => log::error!("Could not extract value: {err}"),
+            };
```

Comment:
I think it's better to exit the application in case of an error instead of all the logging.
Additionally, you should check whether the other parameter is also set when one is set, because it's doesn't make sense to just set one of them.